arm-sql/2015-05-01-preview/swagger/advisors.json

```diff
@@ -0,0 +1,981 @@
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RecommendedActionImpactRecord"
+          }
+        },
+        "observedImpact": {
+          "description": "Gets the observed/actual impact info for this recommended action e.g., Actual CPU gain, Actual Disk Space change",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RecommendedActionImpactRecord"
+          }
+        },
+        "timeSeries": {
+          "description": "Gets the time series info of metrics for this recommended action e.g., CPU consumption time series",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RecommendedActionMetricInfo"
+          }
+        },
+        "linkedObjects": {
+          "description": "Gets the linked objects, if any.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "details": {
+          "description": "Gets additional details specific to this recommended action.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "object"
+          }
+        }
+      }
+    },
+    "RecommendedActionStateInfo": {
+      "description": "Contains information of current state for an Azure SQL Database, Server or Elastic Pool Recommended Action.",
+      "required": [
+        "currentValue"
+      ],
+      "type": "object",
+      "properties": {
+        "currentValue": {
+          "description": "Current state the recommended action is in. Some commonly used states are: Active      -> recommended action is active and no action has been taken yet. Pending     -> recommended action is approved for and is awaiting execution. Executing   -> recommended action is being applied on the user database. Verifying   -> recommended action was applied and is being verified of its usefulness by the system. Success     -> recommended action was applied and improvement found during verification. Pending Revert  -> verification found little or no improvement so recommended action is queued for revert or user has manually reverted. Reverting   -> changes made while applying recommended action are being reverted on the user database. Reverted    -> succesfully reverted the changes made by recommended action on user database. Ignored     -> user explicitly ignored/discarded the recommended action. ",
+          "enum": [
+            "Active",
+            "Pending",
+            "Executing",
+            "Verifying",
+            "PendingRevert",
+            "RevertCancelled",
+            "Reverting",
+            "Reverted",
+            "Ignored",
+            "Expired",
+            "Monitoring",
+            "Resolved",
+            "Success",
+            "Error"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "CurrentState",
+            "modelAsString": true
```

Comment:
This seems like a good place for modelAsString: true since there are so many options you might add more soon. Others seem more like modelAsString: false is better